projects/observability/src/shared/dashboard/widgets/charts/cartesian-widget/cartesian-widget-renderer.component.ts

```diff
@@ -58,13 +58,16 @@ export class CartesianWidgetRendererComponent<TSeriesInterval> extends Interacti
     return this.model.getDataFetcher().pipe(
       tap(fetcher => {
         this.fetcher = fetcher;
+        const defaultInterval =
+          this.model.defaultInterval?.value !== undefined ? this.model.defaultInterval.getDuration() : undefined;
 
         if (this.intervalSupported()) {
           this.intervalOptions = this.buildIntervalOptions();
-          this.selectedInterval = this.getBestIntervalMatch(this.intervalOptions, this.selectedInterval);
+          this.selectedInterval =
```

Comment:
The default behavior belongs inside `getBestIntervalMatch` for two reasons:
1. in this implementation, a default interval is being given precedence over a selected interval
2. That method enforces that the requested interval has an allowed resolution. Even if configured, we need to reject for example a 15s resolution requested at a a 1 week interval.

I think this just looks like
 ```ts
this.selectedInterval = this.getBestIntervalMatch(this.intervalOptions, this.selectedInterval ?? this.model.defaultInterval?.getDuration());
```

Reply:
done